pkg/controller/virtualmachine/virtualmachine.go

```diff
@@ -364,7 +365,12 @@ func (h *Handler) OnVMIDeletion(_ string, vmi *kubevirtv1.VirtualMachineInstance
 
 	vmObj, err := h.vmCache.Get(vmi.Namespace, vmi.Name)
 	if err != nil {
-		return vmi, fmt.Errorf("error fetching vm object for vmi %s/%s: %v", vmi.Namespace, vmi.Name, err)
+		if k8serrors.IsNotFound(err) {
+			// vm is deleted
```

Comment:
@liujiong63 Thanks for the fix.

In the `delete VM case`, could you add one more check on L375, which detects the `vmObj.DeletionTimestamp != nil` and return nil.